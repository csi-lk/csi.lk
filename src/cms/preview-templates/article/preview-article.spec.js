import React from 'react'

import ArticleTemplatePreview from './preview-article'

describe('Preview Template: Home', () => {
  it('renders', () => {
    const props = {
      entry: {
        getIn: jest.fn(() => 'test'),
      },
      widgetFor: jest.fn(() => 'test'),
    }
    const wrapper = shallow(<ArticleTemplatePreview {...props} />)
    expect(wrapper).toMatchSnapshot()
    expect(props.widgetFor).toHaveBeenCalled()
    expect(props.entry.getIn).toHaveBeenCalledTimes(3)
  })
})
