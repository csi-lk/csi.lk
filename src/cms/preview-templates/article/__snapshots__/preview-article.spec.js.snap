// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Template: Home renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleTemplatePreview
    entry={
      Object {
        "getIn": [MockFunction] {
          "calls": Array [
            Array [
              Array [
                "data",
                "title",
              ],
            ],
            Array [
              Array [
                "data",
                "description",
              ],
            ],
            Array [
              Array [
                "data",
                "tags",
              ],
            ],
            Array [
              Array [
                "data",
                "path",
              ],
            ],
          ],
        },
      }
    }
    widgetFor={
      [MockFunction] {
        "calls": Array [
          Array [
            "body",
          ],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "data": Object {
        "markdownRemark": Object {
          "frontmatter": Object {
            "description": "test",
            "tags": "test",
            "title": "test",
          },
          "html": "test",
        },
      },
      "pathContext": Object {
        "path": "test",
      },
      "preview": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "markdownRemark": Object {
            "frontmatter": Object {
              "description": "test",
              "tags": "test",
              "title": "test",
            },
            "html": "test",
          },
        },
        "pathContext": Object {
          "path": "test",
        },
        "preview": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
